tests/lib/azure/vnet.go

```diff
@@ -24,13 +25,21 @@ func NewVnet(f fixture.F) *Vnet {
 
 // AssertExists checks if virtual network exists in the resource group.
 // Fail tests otherwise.
-func (vnet *Vnet) AssertExists(t *testing.T, name string, address string) {
+func (vnet *Vnet) AssertExists(t *testing.T, name string, address, routeTable string) {
 	vnet.f.Retrier.Run(newID("Vnet", "AssertExists", name), func() error {
 		net, err := vnet.client.Get(vnet.f.ResGroupName, name, "")
 		if err != nil {
 			return err
 		}
 
+		if net.VirtualNetworkPropertiesFormat == nil {
+			return errors.New("The field VirtualNetworkPropertiesFormat is nil!")
+		}
+		properties := *net.VirtualNetworkPropertiesFormat
+
+		if properties.AddressSpace == nil {
+			return errors.New("The field AddressSpace is nil!")
+		}
 		addressActual := *net.VirtualNetworkPropertiesFormat.AddressSpace.AddressPrefixes
 		if addressActual != nil {
 			if addressActual[0] != address {
```

Comment:
Also, what if the slice is not nil...but it is empty ? This will explode. Since nil slices are valid empty slices you could use just:

```
if len(gotAddresses) == 0
```

To determine failure, this will contemplate both nil and empty slices ;-)